File: univariate-review/gists/overlaid-distributions-normal-and-t.py
# The normal and t densities for a sample of 7, so 6 degrees of freedom.
import numpy as np
import plotly.graph_objects as go
from scipy import stats

n = 7
z = np.arange(-5, 5.001, 0.01)

fig = go.Figure()
fig.add_scatter(x=z, y=stats.norm.pdf(z), mode="lines",
                name="Normal distribution")
fig.add_scatter(x=z, y=stats.t.pdf(z, df=n - 1), mode="lines",
                line_dash="dash", name=f"t-distribution (df={n - 1})")
fig.update_layout(xaxis_title_text="z",
                  yaxis_title_text="Normal and t-distributions")
fig.show()

File: univariate-review/gists/overlaid-distributions-normal-and-t.R
# Overlay the normal distribution and the t-distribution
# for a sample of n = 7, so 6 degrees of freedom.
n <- 7
z <- seq(-5, 5, 0.01)
prob.norm <- dnorm(z)
prob.t <- dt(z, df=n-1)

plot(z, prob.norm, type="l", ylab="Normal and t-distributions", lwd=2)
lines(z, prob.t, lty=8, lwd=2)  # dashed line
legend(x=1.35, y=0.25, legend=c("Normal distribution", "t-distribution"),
       lty=c(1,8), lwd=c(2,2))
